src/type-translator.ts

```diff
@@ -174,17 +180,38 @@ export class TypeTranslator {
     }
     const alias = this.symbolsToAliasedNames.get(symAlias);
     if (alias) return alias;
-    if (useFqn) {
-      const fqn = this.typeChecker.getFullyQualifiedName(sym);
-      if (!fqn.startsWith(`"`) && !fqn.startsWith(`'`)) {
-        // Non-quoted FQNs mean the name is a global symbol (e.g. from namespace).
-        return this.stripClutzNamespace(fqn);
-      } else {
-        // TODO(martinprobst): Quoted FQNs mean the name is from a module. We still need to scope it
-        // for ambient declarations in externs.
+    if (useFqn && this.isForExterns) {
+      // For regular type emit, we can use TypeScript's naming rules, as they match Closure's name
+      // scoping rules. However when emitting externs files for ambients, naming rules change. As
+      // Closure doesn't support externs modules, all names must be global and use global fully
+      // qualified names. The code below uses TypeScript to convert a symbol to a full qualified
+      // name and then emits that.
+      let fqn = this.typeChecker.getFullyQualifiedName(sym);
+      if (fqn.startsWith(`"`) || fqn.startsWith(`'`)) {
+        // Quoted FQNs mean the name is from a module, e.g. `'path/to/module'.some.qualified.Name`.
```

Comment:
I looked at the TS code, unless I'm mistaken this always refers to the file the name is defined in.